query-engine/black-box-tests/tests/helpers/mod.rs

```diff
@@ -34,12 +34,30 @@ where
     Ok(())
 }
 
-/// Configures the query-engine binary to listen in given port and using the
-/// given dml string as the schema.
-pub(crate) fn query_engine_cmd(dml: &str, port: &str) -> process::Command {
+// Sequentially generates port numbers, which are potentially free. No checks
+// are performed to ensure that nobody actually occupies the port.
+fn generate_free_port() -> u32 {
+    static PORT_NUMBER: Mutex<u32> = Mutex::new(57582);
+    let mut guard = PORT_NUMBER.lock().unwrap();
+    let result = *guard;
+    *guard += 1;
+    result
+}
+
+// Configures the query-engine binary using given DML string as the schema. Returns command to start
+// the query-engine and an HTTP URL to access it.
+pub(crate) fn query_engine_cmd(dml: &str) -> (process::Command, String) {
     let mut cmd = std::process::Command::new(query_engine_bin_path());
-    cmd.env("PRISMA_DML", dml).arg("--port").arg(port).arg("-g");
-    cmd
+    // Clear environment to prevent variables set outside of test affect execution.
+    cmd.env_clear();
+
+    let port = generate_free_port();
+    cmd.env("PRISMA_DML", dml)
+        .arg("--port")
+        .arg(&port.to_string())
```

Comment:
This could potentially be rewritten as
```rust
let ip = "http://0.0.0.0";
let url = format!("{}:{}", ip, port);

cmd.env(..)
  .arg(&url[ip.len()+1..])
```

to avoid extra allocation, but these are tests and I am not too worried about performance here.